packages/next/src/middleware/sitemaps/createSitemaps.ts

```diff
@@ -51,21 +52,54 @@ const parserConfig: Partial<X2jOptions> = {
  * @returns {Response|undefined}
  */
 export async function createRootSitemapIndex(
-  req: NextRequest,
+  req: NextRequest | IncomingMessage,
   normalizedConfig: NormalizedConfig,
+  isMiddleware = true,
 ): Promise<Response | undefined> {
   const { pages, sitemapPathsToIgnore, replaceUrls, wpUrl } = normalizedConfig;
-  const { pathname, origin } = new URL(req.url);
+
+  if (!req.url) {
+    throw new Error('Request object must have URL');
+  }
+
+  let wpSitemapUrl = '';
+  let frontendUrl = '';
+  if (isMiddleware) {
+    const { pathname, origin } = new URL(req.url);
+    frontendUrl = origin;
+    wpSitemapUrl = `${trimSlashes(wpUrl)}/${trimSlashes(pathname)}`;
+  } else {
+    // get rootSitemapPath config param
+    // fetch sitemap from WP
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-argument
+    wpSitemapUrl = `${trimSlashes(wpUrl)}/${trimSlashes(
+      // eslint-disable-next-line @typescript-eslint/no-unsafe-argument
+      normalizedConfig.rootSitemapPath,
```

Comment:
I'm not sure why I missed this during our pair programming but we already have a config setting for the root sitemap path, it's called `sitemapIndexPath`. Let's use this instead as it's one less new option to introduce.